profilingdemo/source/java/src/org/kaaproject/kaa/demo/profiling/ProfilingDemo.java

```diff
@@ -0,0 +1,198 @@
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        LOG.info("--= Press any key to exit =--");
+        try {
+            System.in.read();
+        } catch (IOException e) {
+            LOG.error("IOException was caught - ", e);
+        }
+
+        // Stop the Kaa client and release all the resources which were in use.
+        try {
+            stopKaaClients();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        LOG.info("Profiling demo stopped");
+    }
+
+    public static void startKaaClient(final int index) throws IOException {
+
+        KaaClientProperties kaaClientProperties = new KaaClientProperties();
+        kaaClientProperties.setWorkingDirectory(PROPERTIES_OUT_KEYS_DIR + index);
+
+        // Create the Kaa desktop context for the application.
+        DesktopKaaPlatformContext desktopKaaPlatformContext = new DesktopKaaPlatformContext(kaaClientProperties);
+
+        // Create a Kaa client and add a listener which displays the Kaa client profiling
+        // as soon as the Kaa client is started.
+        final KaaClient kaaClient = Kaa.newClient(desktopKaaPlatformContext, new SimpleKaaClientStateListener() {
+            @Override
+            public void onStarted() {
+                super.onStarted();
+
+                displayConfiguration(index);
+            }
+        }, true);
+        kaaClients.add(kaaClient);
+
+        kaaClient.setProfileContainer(new ProfileContainer() {
+            @Override
+            public PagerClientProfile getProfile() {
+                InnerData data = init().get(index);
+
+                PagerClientProfile pagerClientProfile = new PagerClientProfile();
+                pagerClientProfile.setAudioSupport(data.isAudio());
+                pagerClientProfile.setVibroSupport(data.isVibro());
+                pagerClientProfile.setVideoSupport(data.isVideo());
+
+                return pagerClientProfile;
+            }
+        });
+
+        // Persist profiling in a local storage to avoid downloading it each time the Kaa client is started.
+        kaaClient.setConfigurationStorage(new SimpleConfigurationStorage(desktopKaaPlatformContext, "saved_config.cfg"));
+
+        // Add a listener which displays the Kaa client profiling each time it is updated.
+        kaaClient.addConfigurationListener(new ConfigurationListener() {
+            @Override
+            public void onConfigurationUpdate(PagerConfiguration configuration) {
+                displayConfiguration(index);
+            }
+        });
+
+        // Start the Kaa client and connect it to the Kaa server.
+        kaaClient.start();
+    }
+
+    public static void stopKaaClients() throws IOException {
+        LOG.info("Stopping clients...");
+        for (KaaClient k : kaaClients) {
+            if (k != null)
+                k.stop();
+        }
+    }
+
+    public static void displayConfiguration(int index) {
```

Comment:
Add `synchronized` key-word. `kaaClient`s running in their own threads and call this method simultaneously and this will cause bugs in logs.